### PR TITLE
DC-434: Support flow data in minion gateway

### PR DESCRIPTION
**Pull request by freddy61025**
## Description

- Minion gateway can accept flow payloads 
- Minion gateway can push flows data to Kafka

## Jira link(s)
- https://opennms.atlassian.net/browse/DC-434

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [x] Documentation has been updated as necessary.

### CODE REVIEW

File: minion-gateway/main/src/main/java/org/opennms/miniongateway/grpc/server/rpcrequest/flow/FlowKafkaForwarder.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
```
Comment (patrick-schweizer) [inline minion-gateway/main/src/main/java/org/opennms/miniongateway/grpc/server/rpcrequest/flow/FlowKafkaForwarder.java:4]: 2023? :-D

**Review by patrick-schweizer — APPROVED** [review]
:+1:

**Review by FrancescoBuc86 — APPROVED** [review]

File: minion-gateway/main/src/main/java/org/opennms/miniongateway/grpc/server/rpcrequest/flow/FlowKafkaForwarder.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Forwarder of Flow messages - received via GRPC and forwarded to Kafka.
+ */
+@Component
+public class FlowKafkaForwarder implements MessageConsumer<Message, Message> {
+    public static final String TENANT_ID_HEADER_NAME = "tenant-id";
+    public static final String DEFAULT_TASK_RESULTS_TOPIC = "flow";
```
Comment (cgorantla) [inline minion-gateway/main/src/main/java/org/opennms/miniongateway/grpc/server/rpcrequest/flow/FlowKafkaForwarder.java:56]: nit : `flows` may be better name for the topic ?

**Review by cgorantla — APPROVED** [review]

File: shared-lib/horizon-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/flow/telemetry.proto [hunk 3]
```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+package opennms.sink.flow;
+
+option java_multiple_files = true;
+option java_package = "org.opennms.horizon.grpc.telemetry.contract";
+option java_outer_classname = "TelemetryProtos";
+
+message TelemetryMessage {
+    uint64 timestamp = 1;
+    bytes bytes = 2;
```
Comment (j-white) [inline shared-lib/horizon-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/flow/telemetry.proto:10]: Shouldn't we send something more structured out of the gateway?

Comment (freddy61025) [reply shared-lib/horizon-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/flow/telemetry.proto:10]: Sure we also planning to do it. We are trying to make the whole write path working as soon as possible and optimize them  later.

**Pull request merged**